tests/v3/connection_state_test.c

```diff
@@ -2997,6 +3000,7 @@ static int s_test_mqtt_connection_publish_QoS1_timeout_fn(struct aws_allocator *
         state_test_data);
     ASSERT_TRUE(packet_id_1 > 0);
 
+    uint64_t first_ping_time = connection->next_ping_time;
```

Comment:
It's not threadsafe to reach inside the connection if you're not in its event loop thread.